PyStemmusScope/bmi/implementation.py

```diff
@@ -59,23 +58,32 @@ def load_state(config: dict) -> h5py.File:
     return h5py.File(matfile, mode="a")
 
 
-def get_variable(state: h5py.File, varname: str) -> np.ndarray:
+def get_variable(
+    state: h5py.File, varname: str
+) -> np.ndarray:  # noqa: PLR0911 PLR0912 C901
     """Get a variable from the model state.
 
     Args:
         state: STEMMUS_SCOPE model state
         varname: Variable name
     """
-    if varname == "respiration":
-        return state["fluxes"]["Resp"][0]
+    if varname not in MODEL_VARNAMES:
+        msg = "Unknown variable name"
+        raise ValueError(msg)
+
+    # deviating implemetation:
     elif varname == "soil_temperature":
         return state["TT"][0, :-1]
```

Comment:
I have to modify the size here, otherwise it will not match the soil grid, and the soil_moisture and soil_temperature variables will have different sizes.

> The length of the array might change in stemmus_scope, then this will go wrong.

On a new STEMMUS_SCOPE version, the BMI will also have to be updated. This is already hardcoded.